### PR TITLE
Ability to sign hex string

### PR DESCRIPTION
**Pull request by jstoxrocky**
**What was broken:**
Referencing #1168. Trying to sign a raw hex string (results of a keccak256 hash for example) with `web3.eth.accounts.sign` will throw an `Invalid continuation byte` error. This is because of this line: https://github.com/ethereum/web3.js/blob/dd55daa6cd54a3907457169425bcd5e29bf32ec4/packages/web3-eth-accounts/src/index.js#L219 The above line assumes that any hex string passed to the `hashMessage` function is utf8 encodable. The results of a keccak256 hash are not utf8 encodable (Maybe the number is too large or something. I'm not an expert on bytes and encodings so I may not be saying this correctly!). 

However, the ability to sign a hex string (specifically the results of a keccak256 hash) is important functionality. Say I have a state-channel contract and I want to hash some pre-image values, and then sign the resulting hash. Attempting this with the current implementation throws an error since `msg` is not a utf8 encodable hex string.

```javascript
var msg = web3.utils.soliditySha3(
      {type: 'address', value: some.address}
      {type: 'uint256', value: some.value},
);
var sig = web3.eth.accounts.sign(msg, some.privateKey);
``` 

**How it was fixed:**
Two fixes were made in this PR. 
(1) `web3.eth.accounts.hashMessage` will now break data into Buffers and performs a keccak256 hash on on the Buffer data.
(2) The existing implementation of`web3.eth.accounts.recover` allows users to pass in either the pre-image message or the hashed-message. This is incompatible with allowing hex strings that are not utf8 encodable to be pre-image values. This method was changed to only allow hashed-messages to be passed as arguments. This can be seen as the mirror image of the `web3.eth.accounts.sign` method, which only allows pre-images to be passed as arguments.
To sum up, users should pass pre-images to `sign` and message-hashes to `recover`. I believe that This is in line with how `web3.py` works.

### CODE REVIEW

**Comment by frozeman** [comment]
Thanks for that.
In this case one can only pass in hex data, to be signed, no string directly, is that so in web3.py?

**Comment by jstoxrocky** [comment]
@frozeman You can also pass a string. All the existing tests on regular old strings still pass. They are still in the `test/eth.accounts.sign.js` file. This PR just provides the _added_ functionality to sign hex strings of the form `0x1234...6789` that can't be utf-8 encoded.

Web3.py forces the user to submit a bytes array actually. Strings cannot be passed. The user mist convert them to bytes first. I guess the equivalent of forcing the user to convert their data into a Buffer for Web3.js.

**Comment by frozeman** [comment]
Sorry for the late response, i was on holidays.
You changed the tests to pre-convert from UTF to HEX, so its not working as before actually.
